Sources/MongoKitten/AggregateBuilder.swift

```diff
@@ -0,0 +1,174 @@
+import BSON
+
+public final class SubAggregateBuilder: AggregateBuilder {
+    public var pipeline: [Document] = []
```

Comment:
I'd prefer this implementation to be `private` along with the pipeline removed from the protocol. And just implement `append` in here. It's not like we'll ever change this code again, but I prefer specifying the protocol as minimal as possible for future improvements. So that we don't shoot our future selves in the foot.